src/libcore/fmt/mod.rs

```diff
@@ -610,7 +610,7 @@ pub trait Debug {
 #[rustc_on_unimplemented(
     message="`{Self}` doesn't implement `{Display}`",
     label="`{Self}` cannot be formatted with the default formatter; \
-           try using `:?` instead if you are using a format string",
+           try using `{:#?}` instead if you are using a format string",
```

Comment:
Could you include both:
> try using `{:?}` or `{:#?}` instead if you are using a format string
Or maybe something like:
> try using `{:?}` (or `{:#?}` for extra padding) instead if you are using a format string
I think both are useful in their own right, so I don't think in making one clearer, we should hide the other.